CONTRIBUTING.md

```diff
@@ -1,8 +1,55 @@
-Contributions to cleanlab are welcome. When contributing, please keep the following in mind:
+# Contributing to cleanlab
 
-1. Avoid adding blank lines or spaces to the ends of lines.
-2. Follow PEP-8 coding style.
-3. Always add tests to check the code you added. Tests go in `cleanlab/tests/`.
-4. Submit a pull request (it will automatically run your code and make sure nothing breaks).
+All kinds of contributions to cleanlab are greatly appreciated. If you're not
+looking to write code, submitting a [feature request](#feature-requests) or
+[bug report](#bug-reports) is a great way to contribute. If you want to get
+your hands dirty, you can submit a [patch](#patches), either working on your
+own ideas or addressing [existing issues][issues].
 
-Thanks!
+If you are unsure or confused about anything, please go ahead and submit your
+issue or pull request anyways! We appreciate all contributions, and we'll do
+our best to incorporate your feedback or code into cleanlab.
+
+## Feature Requests
+
+Do you have an idea for an awesome new feature for cleanab? Let us know by
+[submitting a feature request][issue].
+
+If you are inclined to do so, you're welcome to [fork][fork] cleanlab, work on
+implementing the feature yourself, and submit a patch. In this case, it's
+*highly recommended* that you first [open an issue][issue] describing your
+enhancement to get early feedback on the new feature that you are implementing.
+This will help avoid wasted efforts and ensure that your work is incorporated
+into the cleanlab code base.
+
+## Bug Reports
+
+Did something go wrong with cleanlab? Sorry about that! Bug reports are greatly
+appreciated!
+
+When you [submit a bug report][issue], please include as much context as you
+can. This includes information like Python version, cleanlab version, an error
+message or stack trace, and detailed steps to reproduce the bug. The more
+information you can include, the better.
+
+## Patches
+
+Want to write code to help improve cleanlab? Awesome!
+
+If there are [open issues][issues], you're more than welcome to work on those.
+If you have your own ideas, that's great too! In that case, before working on
+substantial changes to the code base, it is *highly recommended* that you first
+[open an issue][issue] describing what you intend to work on.
+
+Any changes to the code base should try to follow the style and coding
+conventions used in the rest of the project.
```

Comment:
Should we ask that:  All code should adhere to the [PEP8 style](https://www.python.org/dev/peps/pep-0008/)?

Reply:
I think standardizing code formatting is a great idea, but just asking people to follow a style is not enough. We should set up:

- An auto-formatter that's as easy to run as `go fmt` / `cargo fmt`, which will guarantee compliance with the style rules
- Checking in CI that code is formatted correctly

Python doesn't have a language-mandated standard for this, and I don't know off the top of my head which is the most popular community project for doing this (maybe [Black](https://black.readthedocs.io)?). I'll look into this and set it up as a separate PR.